docs/source/index.rst

```diff
@@ -1,20 +1,202 @@
+In the :code:`schema` above:
+
+-   :code:`Animal` represents a non-abstract vertex. For relational databases, we think of
+    tables as the non-abstract vertices.
+-   :code:`name` is a **property field** and :code:`out_Animal_LivesIn` is a **vertex field**,
+    which represent an outbound edge to a vertex in the graph. For graph databases,
+    edges can be reflected from the database schema. However, for relational databases edges have
+    to be manually specified. See :doc:`SQL <databases/sql>` for more information.
+
+Query Compilation and Execution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once we have the schema info we can write the following query to get the names of all the animals
+that live in Africa:
+
+.. code:: python
+
+    graphql_query = """
+    {
+        Animal {
+            name @output(out_name: "animal_name")
+            out_Animal_LivesIn {
+                name @filter(op_name: "=", value: ["$continent"])
+            }
+        }
+    }
+    """
+    parameters = {'continent': 'Africa'}
+
+There are a couple of things to notice about queries:
 
+- All queries start with a vertex and branch to other vertices using **vertex fields**.
+- **Directives**, such as :code:`@output` and :code:`@filter`, specify the semantics of a query.
+  :code:`@output` indicates which properties to emit and :code:`@filter` specifies a filter
+  operation.
 
+With the query at hand, we can compile it and get the corresponding results from OrientDB.
 
-Indices and tables
-==================
+.. code:: python
+
+    from graphql_compiler import graphql_to_match
+
+    compilation_result = graphql_to_match(
+        schema, graphql_query, parameters, type_equivalence_hints)
+
+    # Execute query assuming a pyorient client. Other clients may have a different interface.
+    print([result.oRecordData for result in client.query(query)])
+    # [{'animal_name': 'Elephant'}, {'animal_name': 'Lion'}, ...]
+
+What's Next?
+------------
+
+Core Specification
+~~~~~~~~~~~~~~~~~~
+
+.. TODO: We might want to get rid of the definitions section and introduce the terms more
+         naturally.
+
+To learn more about the core specification of the GraphQL query language see:
+
+    - :doc:`Definitions <core_specification/definitions>`, for the definitions of key terms that we
+      use to define the language.
+    - :doc:`Schema Types <core_specification/schema_types>`, for information about the full
+      breadth of schema types that we use to represent database schemas and how to interact
+      with them using GraphQL queries.
+    - :doc:`Query Directives <core_specification/query_directives>`, to learn more about the
+      available directives and how to use them to create powerful queries.
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Core Specification
+   :hidden:
+
+   Definitions <core_specification/definitions>
+   Schema Types <core_specification/schema_types>
+   Query Directives <core_specification/query_directives>
+
+Databases
+~~~~~~~~~
+
```

Comment:
Intro this "to learn more" sentence with a description of why I should care about this Databases section.